### PR TITLE
HIVE-24610 Remove superfluous throws IOException from Context

### PR DESCRIPTION
**Pull request by miklosgergely**
### What changes were proposed in this pull request?
Remove superfluous throws IOException from Context constructor, which may never be thrown.

### Why are the changes needed?
To keep the code clean by not having unused parts in it.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
All the tests are still running fine.

### CODE REVIEW

File: ql/src/test/org/apache/hadoop/hive/ql/exec/tez/TestTezTask.java [hunk 1]
```diff
@@ -196,7 +196,7 @@ public void tearDown() throws Exception {
   }
 
   @Test
-  public void testBuildDag() throws IllegalArgumentException, IOException, Exception {
+  public void testBuildDag() throws IllegalArgumentException, Exception {
```
Comment (kasakrisz) [inline ql/src/test/org/apache/hadoop/hive/ql/exec/tez/TestTezTask.java:199]: Listing `IllegalArgumentException` is not necessary if `Exception` can be thrown.

File: ql/src/test/org/apache/hadoop/hive/ql/exec/tez/TestTezTask.java [hunk 2]
```diff
@@ -217,7 +217,7 @@ public void testBuildDag() throws IllegalArgumentException, IOException, Excepti
   }
 
   @Test
-  public void testEmptyWork() throws IllegalArgumentException, IOException, Exception {
+  public void testEmptyWork() throws IllegalArgumentException, Exception {
```
Comment (kasakrisz) [inline ql/src/test/org/apache/hadoop/hive/ql/exec/tez/TestTezTask.java:220]: Listing `IllegalArgumentException` is not necessary if `Exception` can be thrown.

File: ql/src/test/org/apache/hadoop/hive/ql/exec/tez/TestTezTask.java (hunk 1 above)
Comment (miklosgergely) [reply ql/src/test/org/apache/hadoop/hive/ql/exec/tez/TestTezTask.java]: Nice catch, thanks, fixed.

File: ql/src/test/org/apache/hadoop/hive/ql/exec/tez/TestTezTask.java (hunk 2 above)
Comment (miklosgergely) [reply ql/src/test/org/apache/hadoop/hive/ql/exec/tez/TestTezTask.java]: Fixed.

**Review by kasakrisz — APPROVED** [review]

**Pull request merged**